server/index.js

```diff
@@ -261,7 +258,71 @@ app.delete('/api/edit/:postId', (req, res, next) => {
     .then(result => {
       const data = result.rows;
       if (data) {
-        res.status(200).json(data);
+        return res.status(200).json(data);
+      }
+      throw new ClientError(404, `Cannot find post with postId of ${postId}`);
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/favorite/:postId', (req, res, next) => {
+  const { userId } = req.user;
+  const postId = Number(req.params.postId);
+  if (!Number.isInteger(postId) || postId < 1) {
+    throw new ClientError(400, 'postId must be a positive integer');
+  }
+  const sql = `
+  insert into "favorite" ("userId", "postId", "isFavorite")
+  values ($1, $2, $3)
+  returning*
+  `;
+  const params = [userId, postId, true];
```

Comment:
Here we don't need to add `isFavorite` because the existence of the row is enough.